core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -630,6 +630,11 @@ public int regexpInstr(String value, String regex, int position,
       return matchIndex;
     }
 
+    /** SQL {@code REGEXP_REPLACE} function with 2 arguments. */
+    public String regexpReplace(String s, String regex) {
+      return regexpReplace(s, regex, "", 1, 0, null);
```

Comment:
This is all predicated on regexpReplace doing the right thing.
I guess that is not in the scope of this PR, but all these regular expression languages have slight differences from each other, so it's not obvious to me that regexpReplace implements faithfully what each of these databases expects.

Reply:
You are correct, there may be functional differences between the implementation of `regexp_replace` here and what the engines actually have. I would like to leave that out of scope for this PR.

Reply:
I think that could easily evolve into a 1 year project just by itself.